src/core/mapping/base_mapper.cc

```diff
@@ -941,34 +976,37 @@ void BaseMapper::legate_select_sources(const MapperContext ctx,
   std::vector<std::pair<PhysicalInstance, uint32_t /*bandwidth*/>> band_ranking;
   for (uint32_t idx = 0; idx < sources.size(); idx++) {
     const PhysicalInstance& instance = sources[idx];
-    Memory location                  = instance.get_location();
-    if (location.address_space() == local_node) {
-      if (!all_local) {
-        source_memories.clear();
-        band_ranking.clear();
-        all_local = true;
-      }
-    } else if (all_local)  // Skip any remote instances once we're local
-      continue;
-    auto finder = source_memories.find(location);
-    if (finder == source_memories.end()) {
-      affinity.clear();
-      machine.get_mem_mem_affinity(
-        affinity, location, destination_memory, false /*not just local affinities*/);
-      uint32_t memory_bandwidth = 0;
-      if (!affinity.empty()) {
-        assert(affinity.size() == 1);
-        memory_bandwidth = affinity[0].bandwidth;
-      }
-      source_memories[location] = memory_bandwidth;
-      band_ranking.push_back(std::pair<PhysicalInstance, uint32_t>(instance, memory_bandwidth));
-    } else
-      band_ranking.push_back(std::pair<PhysicalInstance, uint32_t>(instance, finder->second));
+    if (!all_local)  // Skip any remote instances once we're local
```

Comment:
I don't think you want to do this check here. Otherwise once you hit the first local instance, you will ignore all the other local instances that come after, even if they happen to have better bandwidth.

Reply:
And I think the same is true about the two `!all_local` checks below.

Reply:
yes, you are right. I will push the fix shortly.